prometric/src/process.rs

```diff
@@ -0,0 +1,224 @@
+use prometheus::{
+    Gauge, Registry,
+    core::{AtomicU64, GenericGauge},
+};
+use sysinfo::{CpuRefreshKind, MemoryRefreshKind, Pid, ProcessRefreshKind, RefreshKind, System};
+
+type UintGauge = GenericGauge<AtomicU64>;
+
+type UintCounter = GenericGauge<AtomicU64>;
+
+/// A collector for process metrics.
+///
+/// # Metrics
+/// - `process_threads`: The number of OS threads used by the process (Linux only).
+/// - `process_cpu_cores`: The number of logical CPU cores available in the system.
+/// - `process_cpu_usage`: The CPU usage of the process as a percentage.
+/// - `process_max_cpu_freq`: The maximum CPU frequency of all cores in MHz.
+/// - `process_min_cpu_freq`: The minimum CPU frequency of all cores in MHz.
+/// - `process_resident_memory_bytes`: The resident memory of the process in bytes.
+/// - `process_resident_memory_usage`: The resident memory usage of the process as a percentage of
+///   the total memory available.
+/// - `process_start_time_seconds`: The start time of the process in UNIX seconds.
+/// - `process_open_fds`: The number of open file descriptors of the process.
+/// - `process_max_fds`: The maximum number of open file descriptors of the process.
+/// - `process_disk_written_bytes_total`: The total written bytes to disk by the process.
+///
+/// # Example
+/// ```rust
+/// use prometheus::Registry;
+/// use prometric::process::ProcessCollector;
+///
+/// let registry = Registry::new();
+/// let collector = ProcessCollector::new(&registry);
+/// collector.collect();
```

Comment:
so this needs to be manually added to the registry by the user? so it means the user needs to use a custom registry with `AppMetrics::builder().with_registry()`? trying to understand the user flow

Reply:
No actually, I just created a `Default` impl that will register with the default registry. But the option is there to specify a custom registry.

Reply:
ok nice